### PR TITLE
[HPED-270] Add AnchorButton Widget

### PR DESCRIPTION
**Pull request by ldzmitrenka**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -449,6 +449,8 @@ public class DialogWithHeading {
 Used to render hidden inputs in TouchUI dialogs. Exposes properties as described in [Adobe's Granite UI manual on Hidden](https://helpx.adobe.com/experience-manager/6-5/sites/developing/using/reference-materials/granite-ui/api/jcr_root/libs/granite/ui/components/coral/foundation/form/hidden/index.html).
 ###### @Hyperlink
 Used to represent a HTML hyperlinks (<a>) in TouchUI dialogs. Exposes properties as described in [Adobe's Granite UI manual on Hyperlink](https://helpx.adobe.com/experience-manager/6-5/sites/developing/using/reference-materials/granite-ui/api/jcr_root/libs/granite/ui/components/coral/foundation/hyperlink/index.html).
+###### @AnchorButton
+Hyperlink is a component to represent a standard HTML hyperlink (<a>), but to look like a button in TouchUI dialogs. Exposes properties as described in [Adobe's Granite UI manual on AnchorButton](https://helpx.adobe.com/experience-manager/6-5/sites/developing/using/reference-materials/granite-ui/api/jcr_root/libs/granite/ui/components/coral/foundation/anchorbutton/index.html).
```
Comment (liubou-masiuk) [inline README.md:453]: "Hyperlink is a component"... perhaps AnchorButton?

File: aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/anchorbutton/AnchorButton.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+import com.exadel.aem.toolkit.api.annotations.meta.PropertyRendering;
+import com.exadel.aem.toolkit.api.annotations.meta.ResourceType;
+import com.exadel.aem.toolkit.api.annotations.meta.ResourceTypes;
+import com.exadel.aem.toolkit.api.annotations.widgets.common.ElementVariantConstants;
+import com.exadel.aem.toolkit.api.annotations.widgets.hyperlink.HyperlinkIconSize;
+import com.exadel.aem.toolkit.api.annotations.widgets.hyperlink.HyperlinkLinkchecker;
+
+/**
+ * Used to set up
+ * https://helpx.adobe.com/experience-manager/6-5/sites/developing/using/reference-materials/granite-ui/api/jcr_root/libs/granite/ui/components/coral/foundation/anchorbutton/index.html
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/anchorbutton/AnchorButton.java:32]: this link is not clickable. maybe it makes sense to use `<a href="...">` here?

File: aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/anchorbutton/AnchorButton.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    @PropertyRendering(ignoreValues = "false")
+    boolean hideText() default false;
+
+    /**
+     * Maps to the x-cq-linkchecker attribute of this TouchUI dialog component's node
+     * @return One of {@code HyperlinkLinkchecker} values
+     * @see HyperlinkLinkchecker
+     */
+    @PropertyRendering(ignoreValues = "NONE")
+    HyperlinkLinkchecker xCqLinkchecker() default HyperlinkLinkchecker.NONE;
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/anchorbutton/AnchorButton.java:80]: this enum us used in more than one widget. Perhaps it makes sense to mode it to some common package (`com.exadel.aem.toolkit.api.annotations.widgets.common` seems to be a good candidate)

File: aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/anchorbutton/AnchorButton.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+     */
+    String icon() default "";
+
+    /**
+     * Maps to the size of the icon of this TouchUI dialog component's node
+     * @return One of {@code HyperlinkIconSize} values
+     * @see HyperlinkIconSize
+     */
+    @EnumValue
+    HyperlinkIconSize iconSize() default HyperlinkIconSize.S;
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/anchorbutton/AnchorButton.java:94]: Same here

File: aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/anchorbutton/AnchorButton.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+/**
+ * Used to set up
+ * https://helpx.adobe.com/experience-manager/6-5/sites/developing/using/reference-materials/granite-ui/api/jcr_root/libs/granite/ui/components/coral/foundation/anchorbutton/index.html
+ * AnchorButton is a component to represent a standard HTML hyperlink (<a>), but to look like a button in TouchUI dialog
+ */
+@Target(ElementType.FIELD)
+@Retention(RetentionPolicy.RUNTIME)
+@ResourceType(ResourceTypes.ANCHOR_BUTTON)
+@PropertyMapping
+@SuppressWarnings("unused")
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/anchorbutton/AnchorButton.java:39]: Per @smiakchilo , if the component is used in tests (as are most of our components), there's no need to put `@SuppressWarnings("unused")`

File: aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/anchorbutton/Size.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exadel.aem.toolkit.api.annotations.widgets.anchorbutton;
+
+/**
+ * Contains possible values of {@link AnchorButton#size()} property
+ */
+@SuppressWarnings("unused")
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/anchorbutton/Size.java:19]: same here. you can use this enum somewhere in tests and remove this fishy suppression :)

**Review by liubou-masiuk — APPROVED** [review]

**Comment by liubou-masiuk** [comment]
@ldzmitrenka , currently there are test failures for your pr. could you please take a look at them?

File: aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/anchorbutton/Size.java (hunk 6 above)
Comment (ldzmitrenka) [reply aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/anchorbutton/Size.java]: Fixed all, thanks

**Pull request merged**